Application Runner Spike

scottpchow23 opened the pull request:
Demonstrates:

- [x] where to place the component that extends ApplicationRunner in the file tree
- [x] what methods to override
- [x] how to access and list commandline argument names and values
- [x] how to access/alter database from within the `run(String[] args)` method
- [x] how to pass commandline args
- [x] how to modify existing files to redirect effects towards a deployed heroku database

scottpchow23 committed:
sc - Provide a small demo of altering the database from an ApplicationRunner

scottpchow23 committed:
sc - add comment to clarify how to test/use commandline args

pconrad reviewed (approved):
Looks good.   I'll see if I can build on this to add the stuff needed to load into a database.

Rather than use the production database, I'll try doing on a demo on a qa database.

Commit:
pc/sc - add command line runner for startup code

Commit:
pc - refactor command line runner

This refactor adds the ability to get error messages for unrecognized
command options.

It also simplifies adding new ones by limiting what you need to do
to simply adding one element to a hash table, and then
writing one function to process each item

Commit:
pc - document CommandLineRunner
